lib/oscap_tests.pm

```diff
@@ -41,53 +59,367 @@ our $f_vlevel = 'ERROR';
+    elsif ($use_content_type == 2) {
+        download_file_from_https_repo($url, $ds_file_name);
+        # Remove original ds file
+        assert_script_run("rm $f_ssg_sle_ds");
+        # Copy downloaded file to correct location
+        assert_script_run("cp $ds_file_name $f_ssg_sle_ds");
+        record_info("Copied ds file", "Copied file $ds_file_name to $f_ssg_sle_ds");
+    }
+}
+sub replace_xccdf_file {
+    # Replace original xccdf file whith built or downloaded from repository
+    my $self = $_[0];
+    my $xccdf_file_name = $_[1];
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+    # ComplianceAsCode repository master branch
+    if ($use_content_type == 3) {
+        assert_script_run("rm $f_ssg_sle_xccdf");
+        # Copy built file to correct location
+        my $xccdf_local_full_file_path = "$compliance_as_code_path/build/$xccdf_file_name";
+        assert_script_run("cp $xccdf_local_full_file_path $f_ssg_sle_xccdf");
+        record_info("Copied xccdf file", "Copied file $xccdf_local_full_file_path to $f_ssg_sle_xccdf");
+    }
+    # compliance-as-code-compiled
+    elsif ($use_content_type == 2) {
+        download_file_from_https_repo($url, $xccdf_file_name);
+        # Remove original xccdf file
+        assert_script_run("rm $f_ssg_sle_xccdf");
+        # Copy downloaded file to correct location
+        assert_script_run("cp $xccdf_file_name $f_ssg_sle_xccdf");
+        record_info("Copied xccdf file", "Copied file $xccdf_file_name to $f_ssg_sle_xccdf");
+    }
+}
 
+sub replace_ansible_file {
+    # Replace original ansible file whith built or downloaded from repository
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/ansible/";
+    # ComplianceAsCode repository master branch
+    if ($use_content_type == 3) {
+        # Remove original ansible file
+        assert_script_run("rm $full_ansible_file_path");
+        my $ansible_local_full_file_path = "$compliance_as_code_path/build/ansible/$ansible_profile_ID";
+        # Copy built file to correct location
+        assert_script_run("cp $ansible_local_full_file_path $full_ansible_file_path");
+        record_info("Copied ansible file", "Copied file $ansible_local_full_file_path to $full_ansible_file_path");
+        upload_logs("$full_ansible_file_path") if script_run "! [[ -e $full_ansible_file_path ]]";
+    }
+    #  compliance-as-code-compiled
+    elsif ($use_content_type == 2) {
+        download_file_from_https_repo($url, $ansible_profile_ID);
+        # Remove original ansible file
+        assert_script_run("rm $full_ansible_file_path");
+        # Copy downloaded file to correct location
+        assert_script_run("cp $ansible_profile_ID $full_ansible_file_path");
+        record_info("Copied ansible file", "Copied file $ansible_profile_ID to $full_ansible_file_path");
+    }
+    # scap-security-guide
+    elsif ($use_content_type == 1) {
```

Comment:
instead of extensive comments, in general in the code some contants could be created on the top of the file in capital letters to represent those numeric values, kind of like enums in other languagues, leaving 0s and 1s like clear booleans.